File: bigneuron/Zhijiang-load DiademMetric and print result.py
'''
Created on 2016/06/11

@author: Zhijiang-PC
'''

import jpype
import os.path
 
classpath ='C:/vaa3d/DiademMetric.jar'

jpype.startJVM(jpype.getDefaultJVMPath(),"-ea","-Djava.class.path=%s"%(classpath))
ReadSWC_class = jpype.JClass("org.krasnow.cng.data.ReadSWC")
SwcTreeNode_class=jpype.JClass("org.krasnow.cng.domain.SwcTreeNode")

DiademMetric_Class=jpype.JClass("org.krasnow.cng.diadem.DiademMetric")

File_Class= jpype.JClass("java.io.File")
List_Class=jpype.java.util.ArrayList()
t = ReadSWC_class()     
Swc_object=SwcTreeNode_class()

a_object=File_Class("C:/vaa3d/DiademMetric/ExampleGoldStandard.swc")
b_object=File_Class("C:/vaa3d/DiademMetric/ExampleTest.swc")
DiademMetric_object=DiademMetric_Class(a_object,b_object)
a_value =ReadSWC_class.convertSwcToBinaryTreeList(a_object).size();
b_value=DiademMetric_object.getXYThreshold()

jprint = jpype.java.lang.System.out.println
jprint(a_value)
jprint(b_value)
jpype.shutdownJVM()

File: bigneuron/Zhijiang-load DiademMetric and print result.java
package sedcond_test_diadem;

import java.io.File;
import java.util.List;

import org.krasnow.cng.data.ReadSWC;
import org.krasnow.cng.diadem.DiademMetric;
import org.krasnow.cng.domain.SwcTreeNode;

public class second { 
	public static void main(String[] args) throws Throwable { 
		ReadSWC read=new ReadSWC();
		File object=new File("C:/vaa3d/DiademMetric/ExampleGoldStandard.swc");
		File object1=new File("C:/vaa3d/DiademMetric/ExampleTest.swc");
		List<SwcTreeNode> list1=ReadSWC.convertSwcToBinaryTreeList(object);
		
		System.out.println(list1.size());
	    System.getProperty("java.classpath");
		DiademMetric object2=new DiademMetric(object,object1,10);
		
		double score=object2.getXYThreshold();
		System.out.println(score);
		
	} 
}